### PR TITLE
Gives a mechanical nudge to Security to look spiffy

### PR DESCRIPTION
**Pull request by bloons3**
## About The Pull Request
- gives Security a (minor) benefit for wearing their issued gloves
- gives Security a (minor) debuff for wearing insulated gloves
- gives Assistants a (minor) buff for getting their tider outfit together

The mood changes are +/- 1, so nothing major.

image
image
image

## Why It's Good For The Game

This is a slight mechanical nudge for Security players to not immediately rush then wear insulated gloves all round.

## Changelog
:cl:
tweak: Security is incentivized to wear their uniform
tweak: Assistants are rewarded for obtaining their shinies
/:cl:

### CODE REVIEW

**Comment by bloons3** [comment]
The logic for this was stolen from Clown Shoes giving the clown a positive moodlet.

https://github.com/BeeStation/BeeStation-Hornet/blob/19a8e80f38277b3a0338f49445adce76f5bc7ddf/code/modules/clothing/shoes/miscellaneous.dm#L93

**Comment by ghost** [comment]
Why is it tied to glove type and not glove color?

**Comment by bloons3** [comment]
> Why is it tied to glove type and not glove color?

You just know there'd be the one guy who buys insulated gloves then heads to dorms to recolor them.

**Comment by ghost** [comment]
> You just know there'd be the one guy who buys insulated gloves then heads to dorms to recolor them.

That would be the point yes, it doesn't even apply to budget insulated gloves which look identical to insulated gloves. If you want to nerf security for wearing insulated gloves, neither does it apply to combat gloves which are also black (and insulated).

At least have it make some sense or make it a buff only.

**Comment by bloons3** [comment]
> > You just know there'd be the one guy who buys insulated gloves then heads to dorms to recolor them.
> 
> That would be the point yes, it doesn't even apply to budget insulated gloves which look identical to insulated gloves. If you want to nerf security for wearing insulated gloves, neither does it apply to combat gloves which are also black (and insulated).
> 
> At least have it make some sense or make it a buff only.

Lowest common denominator.

File: code/datums/mood_events/generic_negative_events.dm [hunk 1]
```diff
@@ -217,3 +217,7 @@
 
 /datum/mood_event/nanite_sadness/add_effects(message)
 	description = "<span class='warning robot'>+++++++[message]+++++++</span>\n"
+
+/datum/mood_event/sec_insulated_gloves
+	description = "<span class='warning'>I look like some dweeb Assistant.</span>\n"
```
Comment (Archanial) [inline code/datums/mood_events/generic_negative_events.dm]: That sounds awful.

File: code/modules/clothing/gloves/color.dm [hunk 2]
```diff
@@ -8,6 +8,22 @@
 	item_color="yellow"
 	resistance_flags = NONE
 
+/obj/item/clothing/gloves/color/yellow/equipped(mob/user, slot)
+	. = ..()
+	if(slot == SLOT_GLOVES)
+		if(user.mind && user.mind.assigned_role == "Assistant")
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "assistant_insulated_gloves", /datum/mood_event/assistant_insulated_gloves)
+		if(user.mind && user.mind.assigned_role == "Security Officer")
```
Comment (bloons3) [inline code/modules/clothing/gloves/color.dm]:
```suggestion
		if(user.mind && user.mind.assigned_role == "Security Officer" && item_color=="yellow")
```

@githubuser343 would this work?

Comment (ghost) [reply code/modules/clothing/gloves/color.dm]: After a brief look over washing machine and crayon code, yes that's how it seems to work. Similarly for black color.

Comment (bloons3) [reply code/modules/clothing/gloves/color.dm]: I can reason not giving black colored insuls a _buff_ by "These don't feel like my normal gloves".

This PR gives the most minor buff to Security Officers that play by the rules, I don't really want to reward people that cheese it.

File: code/datums/mood_events/generic_negative_events.dm (hunk 1 above)
Comment (francinum) [reply code/datums/mood_events/generic_negative_events.dm]:
```suggestion
	description = "<span class='warning'>I look like an Assistant...</span>\n"
```
Would this be acceptable?

**Comment by francinum** [comment]
I doubt this will change much, but it don't hurt nothin' either.

**Review by francinum — APPROVED** [review]

**Comment by utilbird** [comment]
TG closed it, that's good enough for me

**Pull request merged**